appscale/tools/agents/azure_agent.py

```diff
@@ -0,0 +1,717 @@
+#!/usr/bin/env python
+"""
+This file provides a single class, AzureAgent, that the AppScale Tools can use to
+interact with Microsoft Azure.
+"""
+
+# General-purpose Python library imports
+import adal
+import json
+import os.path
+import shutil
+import time
+
+# Azure specific imports
+from azure.batch.models import OSType
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.mgmt.authorization import AuthorizationManagementClient
+from azure.mgmt.compute import ComputeManagementClient
+from azure.mgmt.compute.models import CachingTypes
+from azure.mgmt.compute.models import DiskCreateOptionTypes
+from azure.mgmt.compute.models import HardwareProfile
+from azure.mgmt.compute.models import ImageReference
```

Comment:
It doesn't look like json, shutil, OSType, AuthorizationManagementClient, or ImageReference are used in this module.